frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -0,0 +1,1495 @@
+	fn from(e: sp_npos_elections::Error) -> Self {
+		ElectionError::NposElections(e)
+	}
+}
+
+impl From<FeasibilityError> for ElectionError {
+	fn from(e: FeasibilityError) -> Self {
+		ElectionError::Feasibility(e)
+	}
+}
+
+/// Errors that can happen in the feasibility check.
+#[derive(RuntimeDebug, Eq, PartialEq)]
+pub enum FeasibilityError {
+	/// Wrong number of winners presented.
+	WrongWinnerCount,
+	/// The snapshot is not available.
+	///
+	/// This must be an internal error of the chain.
+	SnapshotUnavailable,
+	/// Internal error from the election crate.
+	NposElection(sp_npos_elections::Error),
+	/// A vote is invalid.
+	InvalidVote,
+	/// A voter is invalid.
+	InvalidVoter,
+	/// A winner is invalid.
+	InvalidWinner,
+	/// The given score was invalid.
+	InvalidScore,
+	/// The provided round is incorrect.
+	InvalidRound,
+}
+
+impl From<sp_npos_elections::Error> for FeasibilityError {
+	fn from(e: sp_npos_elections::Error) -> Self {
+		FeasibilityError::NposElection(e)
+	}
+}
+
+pub use pallet::*;
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + SendTransactionTypes<Call<Self>>
+	where
+		ExtendedBalance: From<InnerOf<CompactAccuracyOf<Self>>>,
+		ExtendedBalance: From<InnerOf<OnChainAccuracyOf<Self>>>,
+	{
+		type Event: From<Event<Self>>
+			+ Into<<Self as frame_system::Config>::Event>
+			+ IsType<<Self as frame_system::Config>::Event>;
+
+		/// Currency type.
+		type Currency: ReservableCurrency<Self::AccountId> + Currency<Self::AccountId>;
+
+		/// Duration of the unsigned phase.
+		#[pallet::constant]
+		type UnsignedPhase: Get<Self::BlockNumber>;
+		/// Duration of the signed phase.
+		#[pallet::constant]
+		type SignedPhase: Get<Self::BlockNumber>;
+
+		/// The minimum amount of improvement to the solution score that defines a solution as
+		/// "better".
+		#[pallet::constant]
+		type SolutionImprovementThreshold: Get<Perbill>;
+
+		/// The priority of the unsigned transaction submitted in the unsigned-phase
+		type UnsignedPriority: Get<TransactionPriority>;
+		/// Maximum number of iteration of balancing that will be executed in the embedded miner of
+		/// the pallet.
+		type MinerMaxIterations: Get<u32>;
+		/// Maximum weight that the miner should consume.
+		///
+		/// The miner will ensure that the total weight of the unsigned solution will not exceed
+		/// this values, based on [`WeightInfo::submit_unsigned`].
+		type MinerMaxWeight: Get<Weight>;
+
+		/// Something that will provide the election data.
+		type DataProvider: ElectionDataProvider<Self::AccountId, Self::BlockNumber>;
+
+		/// The compact solution type
+		type CompactSolution: codec::Codec
+			+ Default
+			+ PartialEq
+			+ Eq
+			+ Clone
+			+ sp_std::fmt::Debug
+			+ CompactSolution;
+
+		/// Accuracy used for fallback on-chain election.
+		type OnChainAccuracy: PerThing128;
```

Comment:
Would have been nice if this was composed as a part of `Fallback`, because we don't need this unless if `Fallback` is `OnChain`, but I think it will complicate things a bit. 

Also, the fallback is a temporary thing. I only want it to be there, ensuring that while we deploy this module in a dummy manner, if it has a bug, it never triggers the on-chain election as in that case it would case some real harm.